torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Landsat 7 Cloud Cover Assessment Validation Data"""
+import abc
+import glob
+import hashlib
+import os
+from functools import lru_cache
+from typing import Any, Callable, Dict, List, Optional, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from PIL import Image
+from rasterio.crs import CRS
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from .geo import NonGeoDataset, RasterDataset
+from .landsat import Landsat7
+from .utils import BoundingBox, download_url, extract_archive, working_dir
+
+
+class Landsat7Irish(RasterDataset):
+    
+    """ Landsat 7 Cloud Cover Assessment Validation Data.
+
+    The `Landsat 7 Cloud Cover Assessment Validation Data <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset is based on Landsat 7
```

Comment:
```suggestion
class L7Irish(RasterDataset):
    """L7 Irish dataset.

    The `L7 Irish <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset is based on Landsat 7
```
People usually refer to the dataset as L7 Irish. The original masks were produced by Irish et al.